sdk/storage/examples/list_blobs_01.rs

```diff
@@ -0,0 +1,169 @@
+use azure_core::prelude::*;
+use azure_storage::blob::prelude::*;
+use azure_storage::core::prelude::*;
+use futures::stream::StreamExt;
+use std::error::Error;
+use std::num::NonZeroU32;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
+    // First we retrieve the account name and master key from environment variables.
+    let account =
+        std::env::var("STORAGE_ACCOUNT").expect("Set env variable STORAGE_ACCOUNT first!");
+    let master_key =
+        std::env::var("STORAGE_MASTER_KEY").expect("Set env variable STORAGE_MASTER_KEY first!");
+
+    let container_name = std::env::args()
+        .nth(1)
+        .expect("please specify container name as command line parameter");
+
+    let http_client: Arc<Box<dyn HttpClient>> = Arc::new(Box::new(reqwest::Client::new()));
+    let storage_account =
+        StorageAccountClient::new_access_key(http_client.clone(), &account, &master_key)
+            .as_storage_client();
+    let container = storage_account.as_container_client(&container_name);
```

Comment:
Would you say "as_" is more idiomatic in Rust? In most other languages, we use "get_" to get a "sub-client". I see "as_" in Rust more for conversations of sort, or is this because `StorageAccountClient` is all these things as well? We tend to think of getting clients from an account.

Reply:
In general in Rust you use `as_` for a function that creates a new struct but does not take ownership and is cheap to call. `into_` and `to_` are used for expensive conversions that do or do not take ownership (respectively).

The difference between `as_` and `get_` is that, in general, `as_` conveys the idea that the whole struct is being used while with `get_` you receive only a subset of it. 

For example, `std::str::String` defines both [get](https://doc.rust-lang.org/std/string/struct.String.html#method.get) and [as_str](https://doc.rust-lang.org/std/string/struct.String.html#method.as_str). `get` gives back a substring while `as_str` gives back the whole string.

In the case of the storage client we are wrapping the whole storage account so I thought `as_storage_client` was the right name. 
The same applies to `storage_account.as_container_client(container_name)`. In my mind it reads as "create a new container client that includes the whole storage account plus the container name".

Maybe someone else can validate my statements, I am by no means an expert on the matter.

Reply:
Thanks. That's very informative. I guess I hadn't quite recognized that pattern, but it makes sense the way you explain it (and does jive with what I've seen in my limited experience with Rust).

Conceptually, I see what you mean. At least in .NET (and I suspect other language SDKs) we do create new instances of classes that represent a container from an account, and blob from container, but they should (and do, AFAIK) share a pipeline that is really the expensive part (built-in and user-added policies, HTTP client and it's mockable transports, etc.). That seems to be what you're insinuating here: it's still all that the account uses, but with a different set of methods.